edgexfoundry/device-virtual-go
# refactor!: Changed configuration and devices files format to YAML

BREAKING CHANGE: Configuration files are now in YAML format, Default file name is now configuration.yaml

closes #367







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-virtual-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions
make test
make docker
make build
.../cmd$ ./device-virtual -cp -o
see below log msg without error:
```
level=INFO ts=2023-03-22T22:02:59.032284631Z app=device-virtual source=config.go:530 msg="Loaded private configuration from ./res/configuration.yaml"
...
level=INFO ts=2023-03-22T22:02:59.120295827Z app=device-virtual source=devices.go:51 msg="Loading pre-defined devices from /home/v/source/repos/device-virtual-go/cmd/res/devices(1 files found)"
```
## New Dependency Instructions (If applicable)
